iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/HereDocumentImpl.java

```diff
@@ -45,13 +46,12 @@ public Kind getKind() {
 
   @Override
   public List<SyntaxToken> literals() {
-    return List.of(content);
+    throw new UnsupportedOperationException("TODO SONARIAC-579 Remove LiteralList.literals()");
   }
```

Comment:
By throwing an exception in this method, it is stopping rules currently using this method from running, which prevent us from seeing what is the impact of this PR on the ruling execution.
If possible, I would recommend to have a temporarily replacement of this method content like we did for ShellForm and ExecForm, you can very probably just copy/paste the same logic there.

Reply:
👍 Perfectly fine, we didn't cover also as this is temp code that is soon to be removed.